environment/src/main/java/com/toggl/environment/services/calendar/PermissionAwareCalendarService.kt

```diff
@@ -0,0 +1,120 @@
+    private val calendarProjection = arrayOf(
+        CalendarContract.Calendars._ID,
+        CalendarContract.Calendars.CALENDAR_DISPLAY_NAME,
+        CalendarContract.Calendars.ACCOUNT_NAME
+    )
+    private val calendarIdIndex = 0
+    private val calendarDisplayNameIndex = 1
+    private val calendarAccountNameIndex = 2
+
+    private val eventsProjection = arrayOf(
+        CalendarContract.Instances._ID,
+        CalendarContract.Instances.BEGIN,
+        CalendarContract.Instances.END,
+        CalendarContract.Instances.TITLE,
+        CalendarContract.Instances.DISPLAY_COLOR,
+        CalendarContract.Instances.CALENDAR_ID,
+        CalendarContract.Instances.ALL_DAY
+    )
+    private val eventIdIndex = 0
+    private val eventStartDateIndex = 1
+    private val eventEndDateIndex = 2
+    private val eventDescriptionIndex = 3
+    private val eventDisplayColorIndex = 4
+    private val eventCalendarIdIndex = 5
+    private val eventIsAllDayIndex = 6
+
+    override fun getAvailableCalendars(): List<Calendar> {
+        if (!context.permissionToReadCalendarWasGranted())
+            return emptyList()
+
+        val cursor = context.contentResolver.query(
+            CalendarContract.Calendars.CONTENT_URI,
+            calendarProjection,
+            null,
+            null,
+            null
+        )
+        return sequence {
+            cursor?.use {
+                if (it.count <= 0)
+                    return@sequence
+
+                while (it.moveToNext()) {
+                    val id = it.getString(calendarIdIndex)
+                    val displayName = it.getString(calendarDisplayNameIndex)
+                    val accountName = it.getString(calendarAccountNameIndex)
+
+                    yield(Calendar(id, displayName, accountName))
+                }
+            }
+        }.toList()
+    }
+
+    override fun getCalendarEvents(
+        fromStartDate: OffsetDateTime,
+        toEndDate: OffsetDateTime,
+        fromCalendars: List<Calendar>
+    ): List<CalendarEvent> {
+        if (!context.permissionToReadCalendarWasGranted() || fromCalendars.isEmpty())
+            return emptyList()
+
+        val cursor = CalendarContract.Instances.query(
+            context.contentResolver,
+            eventsProjection,
+            fromStartDate.toEpochSecond() * 1000,
+            toEndDate.toEpochSecond() * 1000
+        )
+
+        val calendarIds = fromCalendars.map { it.id }.toSet()
+
+        return sequence {
+            cursor?.use {
+                if (it.count <= 0)
+                    return@sequence
+
+                while (it.moveToNext()) {
+                    val isAllDay = it.getInt(eventIsAllDayIndex) == 1
+                    val calendarId = it.getString(eventCalendarIdIndex)
+                    if (isAllDay || !calendarIds.contains(calendarId))
+                        continue
+
+                    yield(it.nextCalendarItem())
```

Comment:
I always prefer to eliminate `continue`/`break` when you're not secrificing redebility. This is one of those cases IMO.  
```suggestion
                    if (!isAllDay && calendarIds.contains(calendarId))
                        yield(it.nextCalendarItem())
```

Reply:
I agree with Tomas, but I'd wrap this in `{ }` because it's not a return statement